internal/pkg/service/common/httpclient/httpclient.go

```diff
@@ -0,0 +1,107 @@
+// Package httpclient provide HTTP client for API calls with support for telemetry and logging.
+package httpclient
+
+import (
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/keboola/go-client/pkg/client"
+	ddHttp "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/env"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/strhelper"
+)
+
+type Config struct {
+	userAgent   string
+	envs        env.Provider
+	debugWriter io.Writer
+	dumpWriter  io.Writer
+}
+
+type Option func(c *Config)
+
+func WithEnvs(envs env.Provider) Option {
+	return func(c *Config) {
+		c.envs = envs
+	}
+}
+
+func WithUserAgent(v string) Option {
+	return func(c *Config) {
+		c.userAgent = v
+	}
+}
+
+func WithDebugOutput(w io.Writer) Option {
+	return func(c *Config) {
+		c.debugWriter = w
+	}
+}
+
+func WithDumpOutput(w io.Writer) Option {
+	return func(c *Config) {
+		c.dumpWriter = w
+	}
+}
+
+func New(opts ...Option) client.Client {
+	// Apply options
+	conf := Config{userAgent: "keboola-go-client"}
+	for _, o := range opts {
+		o(&conf)
+	}
+
+	// Force HTTP2 transport
+	transport := client.HTTP2Transport()
```

Comment:
Jediny rozdiel v CLI bol, ze nemalo vynuteny HTTP2 transport, iba ho preferovalo.
Myslim, ze ho mozeme rovnako vynutit, ako pri API, je to uz dost dobre vyskusane.

Pri CLI sa okrem toho nenastavuje DD telemetria (tak ako doteraz).